src/mc/protocol.go

```diff
@@ -2,42 +2,61 @@ package mc
 
 // Deal with the protocol specification of Memcached.
 
-import (
-  "errors"
-)
+type MCError struct {
+  Status uint16
+  Message string
+}
+
+func NewMCError(status uint16) *MCError {
+  switch status {
+    case 1:
+      return &MCError{Status:status, Message: ErrNotFound}
+    case 2:
+      return &MCError{Status:status, Message: ErrKeyExists}
+    case 3:
+      return &MCError{Status:status, Message: ErrValueTooLarge}
+    case 4:
+      return &MCError{Status:status, Message: ErrInvalidArgs}
+    case 5:
+      return &MCError{Status:status, Message: ErrValueNotStored}
+    case 6:
+      return &MCError{Status:status, Message: ErrNonNumeric}
+    case 0x20:
+      return &MCError{Status:status, Message: ErrAuthRequired}
+
+    // we only support PLAIN auth, no mechanism that would make use of auth
+    // continue, so make it an error for now for completeness.
+    case 0x21:
+      return &MCError{Status:status, Message: ErrAuthContinue}
+    case 0x81:
+      return &MCError{Status:status, Message: ErrUnknownCommand}
+    case 0x82:
+      return &MCError{Status:status, Message: ErrOutOfMemory}
+    default:
+      return nil
```

Comment:
Perhaps later versions of Go improve on this, but go-1.0.3, the version I use in Travis doesn't like this function not ending with a `return` statement. So please move the default outside of the switch to fix this.